### PR TITLE
Implement [MH3] Vexing Bauble

### PR DESCRIPTION
**Pull request by grimreap124**
https://scryfall.com/card/mh3/212/vexing-bauble

### CODE REVIEW

File: Mage.Sets/src/mage/cards/v/VexingBauble.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * @author grimreap124
+ */
+public final class VexingBauble extends CardImpl {
+
+    public VexingBauble(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.ARTIFACT }, "{1}");
+
+        // Whenever a player casts a spell, if no mana was spent to cast it, counter that spell.
+        this.addAbility(new SpellCastNoManaSpentTriggeredAbility(
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/v/VexingBauble.java:28]: I'm curious why you didn't use a similar way than Roiling Vortex. Apart from the way its text is set up, I find that way cleaner than the new `TriggeredAbility`.

Comment (Susucre) [reply Mage.Sets/src/mage/cards/v/VexingBauble.java:28]: Oh right you probably used Lavinia. I guess it's fine and we'll find how to best refactor them together later on.

Comment (grimreap124) [reply Mage.Sets/src/mage/cards/v/VexingBauble.java:28]: I believe I copied [[Boromir, Warden of the Tower]]

Comment (Susucre) [reply Mage.Sets/src/mage/cards/v/VexingBauble.java:28]: Yeah that's the same as Lavinia. I think I'm okay to merge Bauble as it is, and have opened an issue to refactor the similar effects later on, inbetween set implementation. #12356

**Pull request merged**